### PR TITLE
Standardise cli flags

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added new automated tests to verify correctness of my new code.
   * [ x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the section below.

### Changes description
Introduce common flags to avoid code duplication.

### CODE REVIEW

File: src/snowcli/cli/snowpark/procedure_coverage/clear.py [hunk 1]
```diff
@@ -5,17 +5,12 @@
 
 from snowcli import config, utils
 from snowcli.config import AppConfig
-from snowcli.utils import conf_callback, generate_deploy_stage_name
+from snowcli.utils import check_for_connection, generate_deploy_stage_name
 from snowcli.output.printing import print_db_cursor
 
 from . import app
+from ...sql import ConnectionOption
```
Comment (sfc-gh-pjob) [inline src/snowcli/cli/snowpark/procedure_coverage/clear.py:12]: It doesn't look like the best way for this import.

File: src/snowcli/cli/snowpark/procedure_coverage/report.py [hunk 2]
```diff
@@ -9,16 +9,11 @@
 
 from snowcli import config, utils
 from snowcli.config import AppConfig
-from snowcli.utils import conf_callback, generate_deploy_stage_name
+from snowcli.utils import check_for_connection, generate_deploy_stage_name
 
 from . import app
+from ...sql import ConnectionOption
```
Comment (sfc-gh-pjob) [inline src/snowcli/cli/snowpark/procedure_coverage/report.py:15]: It doesn't look like the best way for this import.

File: src/snowcli/cli/common/flags.py [hunk 3]
```diff
@@ -1 +1,48 @@
...
+    "--account",
+    help="Name assigned to your Snowflake account.",
+)
+
+UserOption = typer.Option(
+    None, "-u", "--username", "--user", help="Username to connect to Snowflake."
+)
+
+PasswordOption = typer.Option(
+    ...,
```
Comment (sfc-gh-pjob) [inline src/snowcli/cli/common/flags.py:32]: Why `...`?

File: src/snowcli/cli/common/flags.py [hunk 4]
```diff
@@ -1 +1,48 @@
+from __future__ import annotations
+
+import typer
+from snowcli.utils import check_for_connection
+
 DEFAULT_CONTEXT_SETTINGS = {"help_option_names": ["-h", "--help"]}
+
```
Comment (sfc-gh-pjob) [inline src/snowcli/cli/common/flags.py:7]: Just to be sure. I know that you have checked checkbox in the PR checklist but I want to confirm that lack of changes in README means that you included all previously used option alternatives for each option.

File: src/snowcli/cli/common/flags.py [hunk 5]
```diff
@@ -1 +1,48 @@
...
+
+ConnectionOption = typer.Option(
+    "dev",
```
Comment (sfc-gh-pjob) [inline src/snowcli/cli/common/flags.py:10]: can't we use `default="dev"`? the same question for other options with `None` default value?

File: src/snowcli/cli/snowpark/procedure_coverage/clear.py (hunk 1 above)
Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/snowpark/procedure_coverage/clear.py:12]: Good catch

File: src/snowcli/cli/snowpark/procedure_coverage/report.py (hunk 2 above)
Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/snowpark/procedure_coverage/report.py:15]: Good catch

File: src/snowcli/cli/common/flags.py (hunk 3 above)
Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/common/flags.py:32]: It's refactor, I haven't used it. But I can change it to None

File: src/snowcli/cli/common/flags.py (hunk 4 above)
Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/common/flags.py:7]: This change does not affect application behaviour

File: src/snowcli/cli/common/flags.py (hunk 5 above)
Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/common/flags.py:10]: > Is it only temporary as long as we use snowsql config?

Yes. We will have to polish this once configuration is managed centrally not per command basis.

**Review by sfc-gh-astus — APPROVED** [review]

File: src/snowcli/cli/common/flags.py (hunk 3 above)
Comment (sfc-gh-pjob) [reply src/snowcli/cli/common/flags.py]: It's refactor but the goal of this refactor is to standardise the flags.
IMO lack of explicit flag names allows to have different flags to pass password value what IMO should be standardised.

I'm unresolving conversation.

File: tests/test_utils.py [hunk 6]
```diff
@@ -168,9 +168,10 @@ def test_generate_streamlit_package_wrapper(self):
             "example_stage", "example_module", False
         )
 
-        assert os.path.exists(result)
+        assert result.exists()
         with open(result, "r") as f:
             assert 'importlib.reload(sys.modules["example_module"])' in f.read()
+        os.remove(result)
```
Comment (sfc-gh-turbaszek) [inline tests/test_utils.py:174]: To avoid side effects

**Review by sfc-gh-pjob — APPROVED** [review]

**Pull request merged**